parkings/anonymization/__init__.py

```diff
@@ -0,0 +1,55 @@
+from django.conf import settings
+from django.db import transaction
+from django.utils import timezone
+
+from ..models import Parking, ParkingCheck, PermitLookupItem
+
+
+def anonymize_parking_registration_numbers():
+    parkings = (
+        Parking.objects.filter(time_end__lt=get_past_time())
+        .exclude(registration_number="")
+        .order_by("time_end")
+    )
+
+    for parking in parkings:
+        with transaction.atomic():
```

Comment:
Spotted one thing I missed on the first round; these atomic transactions don't seem to make any difference as there is only one save operation inside them. I wonder if the idea was to run all of the changes in a transaction (which would probably make sense) and this is just in the wrong place.

Reply:
I don't know what the idea was because I am not the author of this code but I concur that having a transaction.atomic() around a single object save is not necessary here. I can move the transaction clauses outside each for loop, e.g.:
```python
for parking in parkings:
        with transaction.atomic():
            parking.registration_number = ""
            parking.normalized_reg_num = ""
            parking.save(update_fields=["registration_number", "normalized_reg_num"])
```
→
```python
with transaction.atomic():
    for parking in parkings:
            parking.registration_number = ""
            parking.normalized_reg_num = ""
            parking.save(update_fields=["registration_number", "normalized_reg_num"])
```
so then either all of them go through or none of them go through.
Would that be an acceptable solution?

Reply:
Having now talked with both the author of this pull request and @tuomas777 we came to the consensus that removing the ```with transaction.atomic():``` clauses is here is ok.

Reply:
Taken into account in PR #223 (See https://github.com/City-of-Helsinki/parkkihubi/pull/223/commits/6de374eb3df75222cddb6c16db7d9c80968f792d).